jobbergate-api/jobbergate_api/apps/file_validation.py

```diff
@@ -0,0 +1,121 @@
+"""
+Validation methods for the uploaded files.
+"""
+from ast import parse as ast_parse
+from functools import wraps
+from pathlib import PurePath
+from typing import BinaryIO, Callable, Union
+
+from buzz import require_condition
+from jinja2 import Environment
+from loguru import logger
+from yaml import safe_load as yaml_safe_load
+
+
+def get_suffix(filename: str) -> str:
+    """
+    Get the suffix (file extension) from a given filename.
+    """
+    return PurePath(filename).suffix
+
+
+def check_uploaded_file_syntax(file_obj: BinaryIO, filename: str) -> bool:
+    """
+    Check the syntax of a given file.
+    """
+    logger.debug(f"Validating source code on {filename=}")
+    suffix = get_suffix(filename)
+    if suffix in syntax_validation_dispatch:
+        validator = syntax_validation_dispatch[suffix]
+        result = validator(file_obj.read())
+        file_obj.seek(0)
+        return result
+    logger.debug(f"Skipping because {suffix} has no syntax validation")
+    return True
+
+
+SyntaxValidationEquation = Callable[[Union[str, bytes]], bool]
+"""Type alias describing the function signature used to validate file syntax."""
```

Comment:
Why not use a normal comment (using `#`) here?

Reply:
Good catch, solved in [c06c77a](https://github.com/omnivector-solutions/jobbergate/pull/450/commits/c06c77a365b850f51b4a65abe680dcfac9c2cce9)